src/OrleansRuntime/Catalog/Catalog.cs

```diff
@@ -215,15 +215,16 @@ public override void GetObjectData(SerializationInfo info, StreamingContext cont
         /// </summary>
         public Dispatcher Dispatcher { get; }
 
-        public IList<SiloAddress> GetCompatibleSiloList(GrainId grain)
+        public IList<SiloAddress> GetCompatibleSiloList(PlacementTarget target)
         {
             // For test only: if we have silos that are not yet in the Cluster TypeMap, we assume that they are compatible
             // with the current silo
             if (this.config.AssumeHomogenousSilosForTesting)
                 return AllActiveSilos;
 
-            var typeCode = grain.GetTypeCode();
-            var compatibleSilos = GrainTypeManager.GetSupportedSilos(typeCode).Intersect(AllActiveSilos).ToList();
+            var typeCode = target.GrainId.GetTypeCode();
+            var versions = new[] {GrainTypeManager.GetAvailableVersions(target.InterfaceId).Max()};
```

Comment:
When implementing " policies" later, we could return a list of version candidates, not only one. But yes, for now we only use the latest one.